New page: aria-modal (#10605)
mdn/content: files/en-us/web/accessibility/aria/attributes/aria-modal/index.md

---
title: 'aria-modal'
slug: Web/Accessibility/ARIA/Attributes/aria-modal
tags: 
  - Accessibility
  - ARIA
  - ARIA attribute
  - ARIA property
  - aria-modal
  - Reference
---

> The `aria-modal` attribute indicates whether an element is modal when displayed.

Reviewer: Is this in a modal state, the parent element of a modal, or something else?

## Description

> When a dialog is displayed, if the user's interaction is limited to the contents of the dialog, until the dialog loses focus or is no longer displayed, the dialog is considered a "modal dialog". The `aria-modal` attribute is used to indicate the presence of a "modal" element, letting the user know the ability to interact with or access other content on the page requires the modal dialog to be hidden (closed) or otherwise lose focus.

Reviewer: ```suggestion
The `aria-modal` attribute is used to indicate the presence of a "modal" element, letting the user know the ability to interact with, or access other content on the page requires the modal dialog to be closed or otherwise lose focus. Modal dialogs are when content is displayed and the user's interaction is limited to its contents until dismissed. 
```

> When a modal element is displayed, focus is generally placed in the modal. Similarly, assistive technologies (AT) generally navigate to the element unless focus has explicitly been set elsewhere. Depending on how the UX of the modal is coded, users may be 'stuck' in the modal.

Reviewer: ```suggestion
When a modal element is displayed, focus is placed in the modal which navigates assistive technologies (<abbr>AT</abbr>) to the modal's content (unless focus has explicitly been set elsewhere). Modals should have focus "trapped" to its contents until dismissed, to communicate the bounds of the modal and help distinguish it from the rest of the page's content.
```

> In creating modals, ensure the modal can be controlled using only descendants of the modal element. In other words, if a modal dialog has a close button, the button should be a descendant of the dialog, contained with the modal container in the DOM. 

Reviewer: ```suggestion
Ensure the modal can be controlled using only its descendant elements. If a modal dialog has a close button, the button should be a descendant contained with the modal container in the DOM. 
```

> When a modal element is displayed, authors SHOULD mark all other contents as inert (such as "inert subtrees" in HTML) if the ability to do so exists in the host language.

Reviewer: ```suggestion
When a modal element is displayed, authors <strong>should</strong> mark all other contents as inert (such as "inert subtrees" in HTML). 
```

Should we mention the `inert` attribute, it's current lack of support, the polyfil, and the polyfil's limitiations?

When supported, inluding `aria-modal="true"` on a [`dialog"`](/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) or [`alertdialog"`](/en-US/docs/Web/Accessibility/ARIA/Roles/alertdialog_role), removes the requirement of putting [`aria-hidden`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) on background content as the `aria-modal` informs assistive technologies that content outside a dialog is inert. Setting `aria-hidden` on all content outside a dialog to make it inert for AT users is still an option, but no longer required. 

## Example

```html

<div id="backdrop" class="no-scroll">
  <div role="alertdialog"
       aria-modal="true"
       aria-labelledby="dialog_label"
       aria-describedby="dialog_desc">
    <h2 id="dialog_label">
      Confirmation
    </h2>
    <div id="dialog_desc">
      <p>
        Are you sure you want to delete this file?
      </p>
    </div>
    <button type="button" onclick="closeDialog(this)">
      No. Close this popup.
    </button>
    <button type="button" onclick="deleteFile(this)">
      Yes. Delete the file.
    </button>
  </div>
</div>
```

> This partial example includes an alertdialog nested in a full-screen, non-scrollable backdrop. The [`role="alertdialog"`](/en-US/docs/Web/Accessibility/ARIA/Roles/alertdialog_role)	identifies the element that serves as the alert dialog container. The [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) provides the alert dialog an accessible name by referring to the element that provides the alert dialog title. The [`aria-describedby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) attribute gives the alert dialog an accessible description by referring to the alert dialog content that describes the primary message or purpose of the alert dialog.

Reviewer: ```suggestion
This partial example includes an `alertdialog` nested in a full-screen, non-scrollable backdrop. 

The [`role="alertdialog"`](/en-US/docs/Web/Accessibility/ARIA/Roles/alertdialog_role)	identifies the element that serves as the alert dialog container. The [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) provides the alert dialog an accessible name by referring to the element that provides the alert dialog title. The [`aria-describedby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) attribute gives the alert dialog an accessible description by referring to the alert dialog content that describes the primary message or purpose of the alert dialog.

```
